### PR TITLE
chore: move v2 api behind ff

### PR DESCRIPTION
**Pull request by srindom**

### CODE REVIEW

File: packages/medusa/src/loaders/api.ts [hunk 1]
```diff
@@ -25,7 +34,26 @@ export default async ({ app, container, configModule }: Options) => {
   })
 
   app.use(bodyParser.json())
-  app.use("/", routes(container, configModule.projectConfig))
+
+  if (featureFlagRouter.isFeatureEnabled(FeatureFlagUtils.MedusaV2Flag.key)) {
+    // TODO: Figure out why this is causing issues with test when placed inside ./api.ts
```
Comment (olivermrbl) [inline packages/medusa/src/loaders/api.ts:39]: nit: I guess we can remove the todo, if this is no longer causing issues

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/medusa/src/loaders/api.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/api.ts:39]: Some integration tests are testing the route under API to validate the workflows and they have the v2 enabled, to the API is not loaded anymore and the tests fails

Comment (adrien2p) [reply packages/medusa/src/loaders/api.ts:39]: for the unit tests, it seams that in the api `featureFlagRouter` is not available

**Pull request merged**